src/hooks/transactions/useCheckTransactionStatus/useCheckTransactionStatus.ts

```diff
@@ -0,0 +1,197 @@
+  [hash: string]: number;
+}
+
+function useManageFailed() {
+  const dispatch = useDispatch();
+  return ({
+    results,
+    hash,
+    sessionId
+  }: {
+    results: SmartContractResult[];
+    hash: string;
+    sessionId: string;
+  }) => {
+    const resultWithError = results.find(
+      (scResult) => scResult?.returnMessage !== ''
+    );
+
+    dispatch(
+      updateSignedTransactionStatus({
+        transactionHash: hash,
+        sessionId,
+        status: TransactionServerStatusesEnum.fail,
+        errorMessage: resultWithError?.returnMessage
+      })
+    );
+    dispatch(
+      updateSignedTransactions({
+        sessionId,
+        status: TransactionBatchStatusesEnum.fail,
+        errorMessage: resultWithError?.returnMessage
+      })
+    );
+  };
+}
+
+function useManageFailedTransactions() {
+  const dispatch = useDispatch();
+  return (sessionId: string) => {
+    dispatch(
+      updateSignedTransactions({
+        sessionId,
+        status: TransactionBatchStatusesEnum.timedOut
+      })
+    );
+  };
+}
+
+export function useCheckTransactionStatus() {
+  const dispatch = useDispatch();
+  const retriesRef = useRef<RetriesType>({});
+  const timeoutRefs = useRef<string[]>([]);
+  const manageFailedTransactions = useManageFailed();
+  const manageTimedOutTransactions = useManageFailedTransactions();
+  const { pendingTransactionsArray } = useGetPendingTransactions();
+
+  function checkTransactionStatus(
+    getTransactionsByHash?: GetTransactionsByHashesType
+  ) {
+    const pendingBatches = pendingTransactionsArray.filter(
+      ([sessionId, transactionBatch]) => {
+        const isPending =
+          sessionId != null && getIsTransactionPending(transactionBatch.status);
+        return isPending;
+      }
+    );
+    if (pendingBatches.length > 0) {
+      pendingBatches.forEach(([sessionId, transactionBatch]) => {
+        checkBatch({
+          sessionId,
+          transactionBatch,
+          getTransactionsByHash
+        });
+      });
+    }
+  }
+
+  async function checkBatch({
+    sessionId,
+    transactionBatch: { transactions, status, customTransactionInformation },
+    getTransactionsByHash = defaultGetTxByHash
+  }: TransactionStatusTrackerPropsType) {
+    const isPending = sessionId != null && getIsTransactionPending(status);
+    try {
+      if (!isPending || transactions == null) {
+        return;
+      }
+
+      const pendingTransactions = getPendingTransactions(
+        transactions,
+        timeoutRefs.current
+      );
+
+      if (pendingTransactions?.length === 0) {
+        return;
+      }
+      const serverTransactions = await getTransactionsByHash(
+        pendingTransactions
+      );
+      for (const serverTransaction of serverTransactions) {
+        const {
+          hash,
+          status,
+          results,
+          invalidTransaction,
+          pendingResults,
+          hasStatusChanged
+        } = serverTransaction;
+        try {
```

Comment:
I know this is my code, by would you mind moving this to a separate function? Something like "checkPendingTransactionStatus(serverTransaction)"